### PR TITLE
Added equals/hashCode method overrides for a few classes and adding StoreReader interface

### PR DESCRIPTION
**Pull request by sunnywu**
Adding a few equals/hashCode methods for private site data generation logics in https://github.com/IABTechLab/uid2-admin/pull/31

### CODE REVIEW

File: src/main/java/com/uid2/shared/auth/Role.java [hunk 1]
```diff
@@ -10,5 +10,7 @@ public enum Role {
     CLIENTKEY_ISSUER,
     OPERATOR_MANAGER,
     SECRET_MANAGER,
-    ADMINISTRATOR
+    ADMINISTRATOR,
+    //UID2-575 for cronjob to trigger private site data generated
+    PRIVATE_SITE_REFRESHER
```
Comment (sunnywu) [inline src/main/java/com/uid2/shared/auth/Role.java:15]: for cronjob to use later

File: src/main/java/com/uid2/shared/auth/EncryptionKeyAcl.java [hunk 2]
```diff
@@ -17,4 +20,26 @@ public boolean canBeAccessedBySite(int siteId) {
 
     public boolean getIsWhitelist() { return isWhitelist; }
     public Set<Integer> getAccessList() { return accessList; }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (gmsdelmundo-ttd) [inline src/main/java/com/uid2/shared/auth/EncryptionKeyAcl.java:25]: Is there a reason why `isWhitelist` and `accessList` are both package-private and non-final? Can we make them private final?

File: src/main/java/com/uid2/shared/model/EncryptionKey.java [hunk 3]
```diff
@@ -48,4 +50,30 @@ public KeyIdentifier getKeyIdentifier() {
     }
 
     public boolean isExpired(Instant asOf) { return !expires.isAfter(asOf); }
+
+    @Override
+    public boolean equals(Object o) {
+        // If the object is compared with itself then return true
```
Comment (gmsdelmundo-ttd) [inline src/main/java/com/uid2/shared/model/EncryptionKey.java:56]: This seems to be autogenerated comments, could they be removed?

File: src/main/java/com/uid2/shared/model/EncryptionKey.java [hunk 4]
```diff
@@ -48,4 +50,27 @@ public KeyIdentifier getKeyIdentifier() {
...
 
     public boolean isExpired(Instant asOf) { return !expires.isAfter(asOf); }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) return true;
+
+        if (!(o instanceof EncryptionKey)) return false;
+
+        // typecast o to Complex so that we can compare data members
```
Comment (gmsdelmundo-ttd) [inline src/main/java/com/uid2/shared/model/EncryptionKey.java:60]: This one as well

**Review by gmsdelmundo-ttd — APPROVED** [review]

**Pull request merged**